content/blog/how-to-monitor-hpe-compute-ops-management.md

```diff
@@ -0,0 +1,362 @@
+
+For information about how to generate an access token for Compute Ops Management, please refer to [this link ](https://developer.greenlake.hpe.com/docs/greenlake/guides/public/authentication/authentication/)for more details.
+
+Only a few resource metrics are currently supported by HPE Compute Ops Management via the RESTful API, but things will change quickly in the coming months. Today, the only metric available is the carbon footprint report but many other resources are available to create nice Grafana dashboards such as data related to the number of servers, health of servers, service packs, groups, etc. 
+
+# Grafana Infinity plugin
+
+There are several Grafana plugins that support data collection via the REST API (e.g. Infinity, [JSON](https://grafana.com/grafana/plugins/simpod-json-datasource/), [JSON API](https://grafana.com/grafana/plugins/marcusolsson-json-datasource/)) but [Infinity ](https://grafana.com/grafana/plugins/yesoreyeram-infinity-datasource/)has the great advantage of offering an advanced query language that is essential for manipulating JSON data into a suitable format that Grafana can understand. This language is called [UQL](https://sriramajeyam.com/grafana-infinity-datasource/wiki/uql/), Infinity's unstructured query language.
+
+UQL is not simple at first glance but I will provide examples in this blog. With UQL, you can customize the results you need regardless of the json format returned by the API.
+
+A UQL query can be formed with a list of commands joined by |. Most of the time, fields are referenced in double quotes and string values are referenced in single quotes as shown below:
+
+![](/img/2022-10-19-16_33_28-hpe-software-‎-onenote-for-windows-10.png)
+
+The following diagram describes the different components of the solution:
+
+![](/img/2022-10-19-16_15_11-lj-synergy-composable-fabric.pptx-powerpoint.png)
+
+## Pros and Cons about this solution
+
+Pros:
+
+* Lightweight solution as it only requires Grafana and an easily installable plugin
+* Supports collecting metrics from any API
+* Cross-platform support, all components can be installed on Microsoft Windows or Linux.
+
+Cons:
+
+* Cannot create a time series Grafana visualization with non-time series data you may retrieve from an API. This would require the use of a database like Prometheus or InfluxDB
+* Requires in-depth knowledge of the UQL language, API, authentication, and methods.
+
+# Configuration
+
+## Prerequisites
+
+* Grafana must be installed, started, and enabled
+* HPE Compute Ops Management API client credentials are required
+
+## Infinity plugin installation
+
+From an SSH session on the Grafana server, enter:\
+
+> <i>grafana-cli plugins install yesoreyeram-infinity-datasource</i>\
+> Then restart the Grafana service:\
```

Comment:
Something seems strange in this section when I go to the Preview of the blog post:
![image](https://user-images.githubusercontent.com/77496261/196796435-522d0c99-9eb0-4c0f-95e1-a89c532d7055.png)

I don't think those slashes should all be there and I think you meant to hotlink the article, but it didn't work.